utilities/os_utils.py

```diff
@@ -213,30 +214,37 @@ def generate_os_matrix_dict(os_name: str, supported_operating_systems: list[str]
     return os_formatted_list
 
 
-def generate_instance_type_rhel_os_matrix(preferences: list[str]) -> list[dict[str, dict[str, Any]]]:
+def generate_instance_type_os_matrix(os_name: str, preferences: list[str]) -> list[dict[str, dict[str, Any]]]:
     """
-    Generate a list of dictionaries representing the instance type matrix for RHEL OS.
+    Generate a list of dictionaries representing the instance type matrix for a given OS type.
```

Comment:
are we making it generic?

Reply:
yes

Reply:
so this is for non windows os ?

Reply:
Yes, for now it is only for centos and rhel, as we create the windows test we need to see if we can use it or not

Reply:
Yes, but there is no windows/fedora tests yet, so in this current state it works for all of the operating systems tests.
I don't think we should reflect on something we haven't implemented yet.

Reply:
but this is part of utilities which means it is generic and it supports limited os. How will someone know that it's just for rhel and centos. somewhere we should reflect it either in function name or docstrings

Reply:
if it is not generic, please move it locally in infrastructure